runtime/kusama/src/lib.rs

```diff
@@ -1326,6 +1327,97 @@ impl pallet_nomination_pools::Config for Runtime {
 	type MaxPointsToBalance = MaxPointsToBalance;
 }
 
+pub struct RemoveGovV1Storage;
+impl RemoveGovV1Storage {
+	fn get_pallets() -> Vec<&'static str> {
+		vec![
+			"Democracy",
+			"Council",
+			"TechnicalCommittee",
+			"PhragmenElection",
+			"TechnicalMembership",
+			"Treasury",
+		]
+	}
+
+	fn count_keys(prefix: &[u8]) -> u32 {
+		let mut current = prefix.clone().to_vec();
+		let mut counter = 0;
+		while let Some(next) = sp_io::storage::next_key(&current[..]) {
+			if !next.starts_with(prefix) {
+				return counter
+			}
+			counter += 1;
+			current = next;
+		}
+		0
+	}
+}
+impl frame_support::traits::OnRuntimeUpgrade for RemoveGovV1Storage {
+	fn on_runtime_upgrade() -> frame_support::weights::Weight {
+		use frame_support::dispatch::GetDispatchInfo;
+		use sp_runtime::traits::Dispatchable;
+
+		// Create a 'kill_prefix' call for each pallet we wish to remove keys for
+		let mut total_keys = 0;
+		let calls = Self::get_pallets()
+			.iter()
+			.map(|pallet| {
+				let prefix = twox_128(pallet.as_bytes());
+				let subkeys = Self::count_keys(&prefix);
```

Comment:
I would add a new helper similar to https://paritytech.github.io/substrate/master/frame_support/storage/migration/fn.move_pallet.html that is `remove_pallet(prefix: &'static str) -> u32` that removes and counts at the same time. 

In your current code, you are doing the expensive iteration over all keys twice, once for counting and once for deleting.